torch/_inductor/scheduler.py

```diff
@@ -362,6 +380,86 @@ def codegen_originating_info(self, buffer, only_once=True):
         buffer.writelines(out_lines)
         self.written = True
 
+    def get_read_write_buffers_sizes(self) -> int:
+        if isinstance(self, NopKernelSchedulerNode):
+            return 0
+        reads = {dep.name for dep in self.read_writes.reads}
+        writes = {dep.name for dep in self.read_writes.writes}
+
+        def is_materialized(buf):
+            buf_uses = {user.node for user in self.scheduler.name_to_node[buf].users}
+            return len(buf_uses - set(self.snodes)) > 0
+
+        if isinstance(self, FusedSchedulerNode):
+            removed_buffers = {dep for dep in writes if not is_materialized(dep)}
+            writes = writes - removed_buffers
+            reads = reads - removed_buffers
+        node_bytes = 0
+        for buf in reads | writes:
+            if buf in V.graph.name_to_buffer:
+                buf = V.graph.name_to_buffer[buf]
+            elif buf in V.graph.graph_inputs:
+                buf = V.graph.graph_inputs[buf]
+            else:
+                continue
+
+            node_bytes += V.graph.sizevars.size_hint(
+                sympy_product(buf.get_size())
+            ) * get_dtype_size(buf.get_dtype())
+        return node_bytes
+
+    def get_estimated_runtime(self) -> float:
+        layout = None
+        dtype = None
+        if not self.node:
+            assert self.snodes
+            layout = self.snodes[0].node.get_layout()
+            dtype = self.snodes[0].node.get_dtype()
+        else:
+            layout = self.node.get_layout()
+            dtype = self.node.get_dtype()
+
+        if "cuda" != layout.device.type:
+            # default to no reordering based on runtime
+            return 0
+
+        # TODO(xmfan): figure out how to get hardware specs
+        try:
+            from triton.testing import get_dram_gbps
+
+            gpu_memory_bandwidth = get_dram_gbps()
```

Comment:
Move this into utils as well?

Reply:
Not sure I understand, do you mean to alias triton.testing.get_dram_gbps as torch._inductor.utils.get_dram_gbps? or is this referring to the discussion for a compute device interface within pytorch?

I looked into triton's implementation, and it involves calling into hip/cuda C driver: driver.py: https://fburl.com/0vcm10ra, cuda.c: https://fburl.com/8cb6wi6j. There's a lot of code that would need to be copied

Reply:
Yeah I mean to alias it into `torch._inductor.utils`, and ideally eventually abstract it into a compute device interface.